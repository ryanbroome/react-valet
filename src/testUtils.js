import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
