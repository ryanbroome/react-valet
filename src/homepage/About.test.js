import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import About from "./About";
import { MockUserContext } from "../testUtils";

it("About: matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext>
        <About />
      </MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("About: matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext currentUser={null}>
        <About />
      </MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
