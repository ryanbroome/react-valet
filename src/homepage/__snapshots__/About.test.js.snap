// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      What is Park Pilot?
    </h1>
    <p
      class="About-subtitle"
    >
       Free Valet Software for tracking your valet operation
    </p>
    <p
      class="About-text"
    >
      Alright, listen up folks. I've been in the valet and parking game for over two decades, seen it all, and heard every excuse in the book. Let me tell you something straight: if you're still dealing with cash, you might as well be handing out bonuses to the valets for a job well stolen! Yeah, you heard me right. Cash disappears faster than you can say 'Where'd my profits go?' It's like playing a game of 'Find the Missing Bills,' and trust me, it's not a game you want to be playing. I've seen it all – valets reusing tickets, conveniently forgetting to issue them, or pulling off some Houdini-level sleight of hand. But fear not, because I've got a solution that'll keep your profits where they belong – in your pocket. This software right here? It's like having a guardian angel watching over your cash, except this one doesn't take coffee breaks. So, do yourself a favor and let's put an end to the disappearing act. With this software, you'll be laughing all the way to the bank – without any unexpected detours. Remember, in the words of the great De Niro: 'You better not, or else...' Let's keep those profits intact, shall we?
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <h1>
      What is Park Pilot?
    </h1>
    <p
      class="About-subtitle"
    >
       Free Valet Software for tracking your valet operation
    </p>
    <p
      class="About-text"
    >
      Alright, listen up folks. I've been in the valet and parking game for over two decades, seen it all, and heard every excuse in the book. Let me tell you something straight: if you're still dealing with cash, you might as well be handing out bonuses to the valets for a job well stolen! Yeah, you heard me right. Cash disappears faster than you can say 'Where'd my profits go?' It's like playing a game of 'Find the Missing Bills,' and trust me, it's not a game you want to be playing. I've seen it all – valets reusing tickets, conveniently forgetting to issue them, or pulling off some Houdini-level sleight of hand. But fear not, because I've got a solution that'll keep your profits where they belong – in your pocket. This software right here? It's like having a guardian angel watching over your cash, except this one doesn't take coffee breaks. So, do yourself a favor and let's put an end to the disappearing act. With this software, you'll be laughing all the way to the bank – without any unexpected detours. Remember, in the words of the great De Niro: 'You better not, or else...' Let's keep those profits intact, shall we?
    </p>
  </div>
</DocumentFragment>
`;
